src/policy/mod.rs

```diff
@@ -136,28 +136,43 @@ impl<Pk: MiniscriptKey, Ctx: ScriptContext> Liftable<Pk> for Terminal<Pk, Ctx> {
             | Terminal::NonZero(ref sub)
             | Terminal::ZeroNotEqual(ref sub) => sub.node.lift()?,
             Terminal::AndV(ref left, ref right) | Terminal::AndB(ref left, ref right) => {
-                Semantic::Threshold(2, vec![left.node.lift()?, right.node.lift()?])
+                Semantic::Threshold(
+                    2,
+                    vec![Arc::new(left.node.lift()?), Arc::new(right.node.lift()?)],
+                )
             }
             Terminal::AndOr(ref a, ref b, ref c) => Semantic::Threshold(
                 1,
                 vec![
-                    Semantic::Threshold(2, vec![a.node.lift()?, b.node.lift()?]),
-                    c.node.lift()?,
+                    Arc::new(Semantic::Threshold(
+                        2,
+                        vec![Arc::new(a.node.lift()?), Arc::new(b.node.lift()?)],
+                    )),
+                    Arc::new(c.node.lift()?),
                 ],
             ),
             Terminal::OrB(ref left, ref right)
             | Terminal::OrD(ref left, ref right)
             | Terminal::OrC(ref left, ref right)
-            | Terminal::OrI(ref left, ref right) => {
-                Semantic::Threshold(1, vec![left.node.lift()?, right.node.lift()?])
-            }
+            | Terminal::OrI(ref left, ref right) => Semantic::Threshold(
+                1,
+                vec![Arc::new(left.node.lift()?), Arc::new(right.node.lift()?)],
+            ),
             Terminal::Thresh(k, ref subs) => {
-                let semantic_subs: Result<_, Error> = subs.iter().map(|s| s.node.lift()).collect();
-                Semantic::Threshold(k, semantic_subs?)
-            }
-            Terminal::Multi(k, ref keys) | Terminal::MultiA(k, ref keys) => {
-                Semantic::Threshold(k, keys.iter().map(|k| Semantic::Key(k.clone())).collect())
+                let semantic_subs: Result<Vec<Semantic<Pk>>, Error> =
+                    subs.iter().map(|s| s.node.lift()).collect();
+                let semantic_subs = semantic_subs?
+                    .into_iter()
+                    .map(|sub| Arc::new(sub))
```

Comment:
In 9de17fccd07ce6bcfdb79ad6816c05acc34fb5bc:

This map should just be `map(Arc::new)`. (Though I assume you're going to remove this in a later commit, since otherwise clippy would've complained about it.) Same in many other cases.

Reply:
We don't run clippy in CI :(

Reply:
I fixed them all, I think.